nat-lab/tests/telio.py

```diff
@@ -625,6 +626,37 @@ async def on_stderr(stdout: str) -> None:
                             container_port,
                         ])
 
+                    if (
+                        isinstance(self.get_router(), LinuxRouter)
+                        and host_os == "Linux"
+                    ):
+                        pids = ""
+                        # Find pid of remote libtelio instance, accounting for the fact that it may take some time for it to be available
+                        while len(pids) == 0:
+                            await asyncio.sleep(0.25)
+                            pids = await execute_and_get_stdout(
+                                self._connection, ["pgrep", "-f", "libtelio_remote.py"]
+                            )
+                        pid = pids.splitlines()[0]
+                        # Get all bound sockets
+                        fds = await execute_and_get_stdout(
+                            self._connection, ["netstat", "-tulnp"]
+                        )
+                        # Get the socket address for the previously found pid
+                        port_regex = re.compile(
+                            r"([0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}:[0-9]{1,5}).*\s"
+                            + pid
+                            + "/python"
+                        )
+                        sockets = re.findall(port_regex, fds)
+                        if len(sockets) == 0:
+                            print(
+                                f"Pyro5 on node {self._node.name} has not bound any sockets"
+                            )
+                        object_uri = f"PYRO:{object_name}@{sockets[0]}"
```

Comment:
This method of figuring out listening port very heavily relies that the process creates only single socket, at least at the time of querying. While it is probably true at the time, it introduces this "hidden" requirement for `libtelio_remote.py` which is very easy to break in non-trivial ways :/.

IMHO it would be much nicer if `libtelio_remote.py` would report listening port chosen by itself, or would use a range which is safe(r) to use (e.g. 0-1024), as OS (in nat-lab configuration) is guaranteed not to select those ports inside of containers.